src/lib/finishPollMessage.ts

```diff
@@ -26,33 +43,70 @@ export async function finishPollMessage({ data, read, persistence, modify }: {
     const poll = await getPoll(String(data.message.id), read);
 
     if (!poll) {
-        throw new Error('no such poll');
+        throw new Error("no such poll");
     }
 
     if (poll.finished) {
-        throw new Error('this poll is already finished');
+        throw new Error("this poll is already finished");
     }
 
     if (poll.uid !== data.user.id) {
-        throw new Error('You are not allowed to finish the poll'); // send an ephemeral message
+        throw new Error("You are not allowed to finish the poll"); // send an ephemeral message
     }
 
     try {
         await finishPoll(poll, { persis: persistence });
 
-        const message = await modify.getUpdater().message(data.message.id as string, data.user);
+        const message = await modify
+            .getUpdater()
+            .message(data.message.id as string, data.user);
         message.setEditor(message.getSender());
 
         const block = modify.getCreator().getBlockBuilder();
 
-        const showNames = await read.getEnvironmentReader().getSettings().getById('use-user-name');
-
-        createPollBlocks(block, poll.question, poll.options, poll, showNames.value);
+        const showNames = await read
+            .getEnvironmentReader()
+            .getSettings()
+            .getById("use-user-name");
+        const wordcloudAPI = await read
+            .getEnvironmentReader()
+            .getSettings()
+            .getById("wordcloud-api");
+        createPollBlocks(
+            block,
+            poll.question,
+            poll.options,
+            poll,
+            showNames.value,
+            wordcloudAPI.value
+        );
 
         message.setBlocks(block);
 
-        return modify.getUpdater().finish(message);
+        modify.getUpdater().finish(message);
+
+        if (poll.wordcloud && wordcloudAPI.value) {
+            let wordList = [] as string[];
+            poll.options.map((option, index) => {
+                wordList = wordList.concat(
+                    Array(poll.votes[index].quantity).fill(option)
+                );
+            });
+            const attachment = <IMessageAttachment>{
+                imageUrl: `${wordcloudAPI.value}?text=${wordList.join(" ")}`,
+            };
+            const wordCloudBuilder = modify
+                .getCreator()
+                .startMessage()
+                .setUsernameAlias(data.user.username)
+                .setRoom(<IRoom>{ ...data.room })
+                .addAttachment(attachment);
+
+            return modify.getCreator().finish(wordCloudBuilder);
+        }
+        return;
     } catch (e) {
-        console.error('Error', e);
+        console.error("Error", e);
```

Comment:
We should always use app level logger i.e. `ILogger` interface for logging any errors in the app. All console logs gets logged on the server logs not on the app logs.